Tests/SecureXPCTests/XPCEncoder/Array encoding tests.swift

```diff
@@ -0,0 +1,232 @@
+		
+		// These don't have regular equality, so we'll check them seperately.
+		let nans = try encode([Float.nan, Float.signalingNaN])
+		XCTAssertEqual(xpc_array_get_count(nans), 2)
+		XCTAssert(xpc_array_get_double(nans, 0).isNaN)
+		XCTAssert(xpc_array_get_double(nans, 1).isNaN)
+	}
+	
+	func testEncodes_arrayOf_Doubles_asArrayOf_XPCDoubles() throws {
+		let doubles: [Double] = [
+			-Double.infinity,
+			-Double.greatestFiniteMagnitude,
+			-123,
+			-0.0,
+			 0.0,
+			 123,
+			 Double.greatestFiniteMagnitude,
+			 Double.infinity
+		]
+		
+		try assert(doubles, encodesEqualTo: createXPCArray(from: doubles, using: { xpc_double_create(Double($0)) }))
+		
+		// These don't have regular equality, so we'll check them seperately.
+		let nans = try encode([Double.nan, Double.signalingNaN])
+		XCTAssertEqual(xpc_array_get_count(nans), 2)
+		XCTAssert(xpc_array_get_double(nans, 0).isNaN)
+		XCTAssert(xpc_array_get_double(nans, 1).isNaN)
+	}
+	
+	// MARK: Misc. types
+	
+	func testEncodes_arrayOf_Bools_asArrayOf_XPCBools() throws {
+		let bools = [false, true]
+		try assert(bools, encodesEqualTo: createXPCArray(from: bools, using: { xpc_bool_create($0) }))
+	}
+	
+	func testEncodes_arrayOf_Data_asArrayOf_XPCData() throws { // 💾
+		throw XCTSkip("Support for Data is not implemented, the contents just end up as a String.")
+		
+		let data = [Data(), "Hello, world!".data(using: .utf8)!]
+		try assert(data, encodesEqualTo: createXPCArray(from: data, using: { datum in
+			datum.withUnsafeBytes { bufferP in xpc_data_create(bufferP.baseAddress, bufferP.count) }
+		}))
+	}
+	
+	func testEncodes_arrayOf_Dates_asArrayOf_XPCDates() throws { // 📆
+		throw XCTSkip("Support for Date is not implemented, the contents just end up as a String.")
+		
+		let dates: [Date] = [.distantPast, .init(), .distantFuture]
+		
+		try assert(dates, encodesEqualTo: createXPCArray(from: dates, using: { date in
+			xpc_date_create(Int64(date.timeIntervalSince1970))
+		}))
+	}
+	
+	func testEncodes_arrayOf_Strings_asArrayOf_XPCStrings() throws {
+		let strings = ["", "Hello, world!"]
+		try assert(strings, encodesEqualTo: createXPCArray(from: strings, using: { str in
+			// Does xpc_string_create copy the string? This would be a dangling pointer otherwise.
+			str.withCString(xpc_string_create)
+		}))
+	}
+	
+	func testEncodes_arrayOf_UUIDs_asArrayOf_XPCUUIDs() throws {
+		throw XCTSkip("Support for UUID is not implemented, the contents just end up as a String.")
+		
+		let uuids = [UUID(), UUID(), UUID()]
+		try assert(uuids, encodesEqualTo: createXPCArray(from: uuids, using: { uuid in
+			var uuidByteTuple = uuid.uuid
+			
+			return withUnsafePointer(to: &uuidByteTuple) { uuidByteTupleP in
+				uuidByteTupleP.withMemoryRebound(to: UInt8.self, capacity: 16) { uuidP in xpc_uuid_create(uuidP) }
+			}
+		}))
+	}
+	
+	func testEncodes_arrayOf_nils() throws {
+		// Signed integers
+		let   ints: [Optional<Int  >] = [.min, -123, 0, 123, .max, nil]
```

Comment:
No, I wouldn't never typically write it like this, but in this case, it fits this tabular format a bit better, IMO (because you can't have a space, like `Int ?`). Compare:

```swift
let   ints: [Int?  ] = [.min, -123, 0, 123, .max, nil]
let  int8s: [Int8? ] = [.min, -123, 0, 123, .max, nil]
let int16s: [Int16?] = [.min, -123, 0, 123, .max, nil]
let int32s: [Int32?] = [.min, -123, 0, 123, .max, nil]
let int64s: [Int64?] = [.min, -123, 0, 123, .max, nil]
```

Reply:
Ah, I see. Can't say I have a strong opinion on this, was just unused to seeing it written as such.